src/axes/timeAxis.ts

```diff
@@ -392,9 +392,14 @@ export module Axes {
       let tickLabelsEnter = tickLabels.enter().append("g").classed(Axis.TICK_LABEL_CLASS, true);
       tickLabelsEnter.append("text");
       let xTranslate = (this._tierLabelPositions[index] === "center" || config.step === 1) ? 0 : this.tickLabelPadding();
-      let yTranslate = this.orientation() === "bottom" ?
-          d3.sum(this._tierHeights.slice(0, index + 1)) - this.tickLabelPadding() :
-          this.height() - d3.sum(this._tierHeights.slice(0, index)) - this.tickLabelPadding();
+      let yTranslate: number;
+      if (this.orientation() === "bottom") {
+        yTranslate = d3.sum(this._tierHeights.slice(0, index + 1)) - this.tickLabelPadding();
+      } else if (this._tierLabelPositions[index] === "center") {
+        yTranslate = this.height() - d3.sum(this._tierHeights.slice(0, index)) - this.tickLabelPadding() - this._maxLabelTickLength();
```

Comment:
For axis with **bottom** orientation, each tier should be like
_____________
tick marks (if `tierLabelPostion` is `center`)
tickLabelPadding
LABEL
tickLabelPadding
______________

The height of each label's bottom line is the `height of next tier` - `tierLabelPdding()`. 


For axis with **top** orientation, each tier should be like
____________________

tickLabelPadding
LABEL
tickLabelPadding
tick marks(if `tierLabelPostion` is `center`)
____________________

The height of each label's bottom line is the tier's bottom line -  tickLabelPadding - the length of tickmarks (if the `tierLabelPostion` is `center`)